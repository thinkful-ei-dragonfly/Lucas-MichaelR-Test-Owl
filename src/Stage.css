.Stage {
  background: #404853;
  width: 75%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: 100vh;
}
