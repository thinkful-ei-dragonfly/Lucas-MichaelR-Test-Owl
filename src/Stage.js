import React from 'react';
import './Stage.css'
import User from './User';

function Stage(props) {
  let users = props.users.map(user => <User key={user.id} display={'onStage'} avatar={user.avatar} name={user.name} onStage={user.onStage}/>);

  return <div className="Stage">
    {users}
  </div>
}

export default Stage;
