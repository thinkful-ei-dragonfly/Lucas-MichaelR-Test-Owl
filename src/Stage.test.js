import React from 'react';
import renderer from 'react-test-renderer';

import Stage from './Stage';

import STORE from './store';

const stageUsers = STORE.participants.filter(participant => participant.onStage === true);

describe('<Stage />', () => {
  it('renders without crashing [snapshot test]', () => {
    // Render this component as JSON
    const tree = renderer.create(<Stage users={stageUsers}></Stage>).toJSON();
    // Checks wether this matches the previous snapshot
    // The previous snapshot will be stored in __snapshots__/Stage .test.js
    expect(tree).toMatchSnapshot();
  })
})