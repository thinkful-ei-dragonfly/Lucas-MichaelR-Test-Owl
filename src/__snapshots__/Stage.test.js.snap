// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage /> renders without crashing [snapshot test] 1`] = `
<div
  className="Stage"
>
  <div
    className="stage-user"
  >
    <div
      className="stage-user-info"
    >
      <p>
        <span
          className="stage-user-name"
        >
          Koren Templeton
        </span>
      </p>
      <span
        className="stage-icons"
      >
        <span
          className="volume"
        >
          <span
            className="lnr lnr-volume-high"
          />
        </span>
        <span
          className="search"
        >
          <span
            className="lnr lnr-magnifier"
          />
        </span>
      </span>
    </div>
    <img
      alt="Koren Templeton profile photo"
      className="stage-img"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stage-user"
  >
    <div
      className="stage-user-info"
    >
      <p>
        <span
          className="stage-user-name"
        >
          Ashla Attwool
        </span>
      </p>
      <span
        className="stage-icons"
      >
        <span
          className="volume"
        >
          <span
            className="lnr lnr-volume-high"
          />
        </span>
        <span
          className="search"
        >
          <span
            className="lnr lnr-magnifier"
          />
        </span>
      </span>
    </div>
    <img
      alt="Ashla Attwool profile photo"
      className="stage-img"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
